packages/paypal-commerce-integration/src/components/PayPalCommercePaymentMethodComponent.tsx

```diff
@@ -57,19 +68,47 @@ const PayPalCommercePaymentMethodComponent: FunctionComponent<
         }
     };
 
+    const renderButton = useCallback(() => {
+        if (currentInstrument && !shouldConfirmInstrument) {
+            paymentForm.hidePaymentSubmitButton(method, false);
+            hasPayPalButton.current = false;
+        } else {
+            paymentForm.hidePaymentSubmitButton(method, true);
+
+            if (!hasPayPalButton.current && renderButtonRef.current) {
+                setTimeout(() => {
+                    renderButtonRef.current?.();
+
+                    hasPayPalButton.current = true;
+                }, 0);
+            }
+        }
+    }, [currentInstrument]);
+
+    useEffect(() => {
+        renderButton();
+    }, [renderButton, renderButtonRef.current]);
+
     useEffect(() => {
         void validateButton();
     }, [termsValue]);
 
+    useEffect(() => {
+        fieldsValuesRef.current = {
+            shouldSaveInstrument: shouldConfirmInstrument || Boolean(shouldSaveInstrument),
+        };
+    }, [shouldSaveInstrument, shouldConfirmInstrument]);
+
     const initializePayment = async () => {
         try {
             await checkoutService.initializePayment({
                 gatewayId: method.gateway,
                 methodId: method.id,
                 [providerOptionsKey]: {
                     container: '#checkout-payment-continue',
-                    onRenderButton: () => {
-                        paymentForm.hidePaymentSubmitButton(method, true);
+                    shouldRenderPayPalButtonOnInitialization: false,
```

Comment:
Should this be false for cases if vaulting feature is disabled or if a guest wants to purchase something with this payment method?

Reply:
It should be `false` for both because of the logic of appearing button

Reply:
I don't really sure. may I ask you to record a proof video with guest flow?

Reply:
You're right. I am on my way to fixing it. Thank you

Reply:
I don't understand what was fixed..

Reply:
it should be false only when BC merchant go through checkout steps and has vaulted instruments otherwise it should be true in my opinion.